### PR TITLE
[lgtm] Fix LGTM build on 202012 branch

### PR DESCRIPTION
**Pull request by kcudnik**
**What I did**

**Why I did it**

**How I verified it**

**Details if related**

### CODE REVIEW

**Comment by nazariig** [comment]
@kcudnik FYI. https://github.com/Azure/sonic-swss/pull/1868

File: lgtm.yml [hunk 1]
```diff
@@ -24,18 +24,21 @@ extraction:
...
-      - dpkg-deb -x libsaimetadata-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
-      - dpkg-deb -x libsaivs_1.0.0_amd64.deb $LGTM_WORKSPACE
-      - dpkg-deb -x libsaivs-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
+      - git clone --recursive https://github.com/Azure/sonic-sairedis; pushd sonic-sairedis; git checkout origin/202012; ./autogen.sh; DEB_BUILD_OPTIONS=nocheck SWSS_COMMON_INC="$LGTM_WORKSPACE/usr/include" SWSS_COMMON_LIB="$LGTM_WORKSPACE/usr/lib/x86_64-linux-gnu" fakeroot debian/rules binary-syncd-vs; popd
+      - dpkg-deb -vx libsairedis_1.0.0_amd64.deb $LGTM_WORKSPACE
+      - dpkg-deb -vx libsairedis-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
+      - dpkg-deb -vx libsaimetadata_1.0.0_amd64.deb $LGTM_WORKSPACE
+      - dpkg-deb -vx libsaimetadata-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
+      - dpkg-deb -vx libsaivs_1.0.0_amd64.deb $LGTM_WORKSPACE
+      - dpkg-deb -vx libsaivs-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
```
Comment (nazariig) [inline lgtm.yml:38]: @kcudnik do we need to have `dpkg-deb -vx` in `master`?

Comment (kcudnik) [reply lgtm.yml:38]: done

File: lgtm.yml [hunk 2]
```diff
@@ -24,18 +24,21 @@ extraction:
...
       - dpkg-deb -x libswsscommon_1.0.0_amd64.deb $LGTM_WORKSPACE
       - dpkg-deb -x libswsscommon-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
-      - git clone --recursive https://github.com/Azure/sonic-sairedis; pushd sonic-sairedis; ./autogen.sh; DEB_BUILD_OPTIONS=nocheck SWSS_COMMON_INC="$LGTM_WORKSPACE/usr/include" SWSS_COMMON_LIB="$LGTM_WORKSPACE/usr/lib/x86_64-linux-gnu" fakeroot debian/rules binary-syncd-vs; popd
-      - dpkg-deb -x libsairedis_1.0.0_amd64.deb $LGTM_WORKSPACE
-      - dpkg-deb -x libsairedis-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
-      - dpkg-deb -x libsaimetadata_1.0.0_amd64.deb $LGTM_WORKSPACE
-      - dpkg-deb -x libsaimetadata-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
-      - dpkg-deb -x libsaivs_1.0.0_amd64.deb $LGTM_WORKSPACE
-      - dpkg-deb -x libsaivs-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
+      - git clone --recursive https://github.com/Azure/sonic-sairedis; pushd sonic-sairedis; git checkout origin/202012; ./autogen.sh; DEB_BUILD_OPTIONS=nocheck SWSS_COMMON_INC="$LGTM_WORKSPACE/usr/include" SWSS_COMMON_LIB="$LGTM_WORKSPACE/usr/lib/x86_64-linux-gnu" fakeroot debian/rules binary-syncd-vs; popd
```
[the remark this answers is not in the document]
Comment (kcudnik) [reply lgtm.yml]: done

File: lgtm.yml [hunk 3]
```diff
@@ -24,18 +24,21 @@ extraction:
       - dh-exec
       - doxygen
       - graphviz
+      - autoconf-archive
     after_prepare:
-      - git clone https://github.com/Azure/sonic-swss-common; pushd sonic-swss-common; ./autogen.sh; fakeroot dpkg-buildpackage -us -uc -b; popd
+      - git clone https://github.com/Azure/sonic-swss-common; pushd sonic-swss-common; git checkout origin/202012; ./autogen.sh; fakeroot dpkg-buildpackage -us -uc -b; popd
```
[the remark this answers is not in the document]
Comment (kcudnik) [reply lgtm.yml]: done

File: lgtm.yml (hunk 1 above)
Comment (nazariig) [reply lgtm.yml:38]: @kcudnik got it

File: lgtm.yml [hunk 4]
```diff
@@ -24,18 +24,21 @@ extraction:
...
       - dpkg-deb -x libswsscommon-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
-      - git clone --recursive https://github.com/Azure/sonic-sairedis; pushd sonic-sairedis; ./autogen.sh; DEB_BUILD_OPTIONS=nocheck SWSS_COMMON_INC="$LGTM_WORKSPACE/usr/include" SWSS_COMMON_LIB="$LGTM_WORKSPACE/usr/lib/x86_64-linux-gnu" fakeroot debian/rules binary-syncd-vs; popd
-      - dpkg-deb -x libsairedis_1.0.0_amd64.deb $LGTM_WORKSPACE
-      - dpkg-deb -x libsairedis-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
-      - dpkg-deb -x libsaimetadata_1.0.0_amd64.deb $LGTM_WORKSPACE
-      - dpkg-deb -x libsaimetadata-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
-      - dpkg-deb -x libsaivs_1.0.0_amd64.deb $LGTM_WORKSPACE
-      - dpkg-deb -x libsaivs-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
+      - git clone -b 202012 --recursive https://github.com/Azure/sonic-sairedis; pushd sonic-sairedis; ./autogen.sh; DEB_BUILD_OPTIONS=nocheck SWSS_COMMON_INC="$LGTM_WORKSPACE/usr/include" SWSS_COMMON_LIB="$LGTM_WORKSPACE/usr/lib/x86_64-linux-gnu" fakeroot debian/rules binary-syncd-vs; popd
+      - dpkg-deb -vx libsairedis_1.0.0_amd64.deb $LGTM_WORKSPACE
```
Comment (nazariig) [inline lgtm.yml:33]: @kcudnik do we need to remove this debug stuff: `dpkg-deb -vx`?

File: lgtm.yml [hunk 5]
```diff
@@ -24,18 +24,21 @@ extraction:
...
+      - dpkg-deb -vx libsairedis-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
+      - dpkg-deb -vx libsaimetadata_1.0.0_amd64.deb $LGTM_WORKSPACE
+      - dpkg-deb -vx libsaimetadata-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
+      - dpkg-deb -vx libsaivs_1.0.0_amd64.deb $LGTM_WORKSPACE
+      - dpkg-deb -vx libsaivs-dev_1.0.0_amd64.deb $LGTM_WORKSPACE
     configure:
       command:
       - ./autogen.sh
+      - ls -al $LGTM_WORKSPACE/usr/include/sai || true ; ls -al /usr/include/sai || true
+      - find / -name sai.h -ls || true
```
Comment (nazariig) [inline lgtm.yml:43]: @kcudnik do we need to remove this debug stuff?

File: orchagent/main.cpp [hunk 6]
```diff
@@ -33,6 +33,8 @@ extern "C" {
 using namespace std;
 using namespace swss;
 
+
+
```
Comment (nazariig) [inline orchagent/main.cpp:37]: @kcudnik do we need to remove this debug stuff?

Comment (kcudnik) [reply orchagent/main.cpp:37]: removed

File: lgtm.yml (hunk 5 above)
Comment (kcudnik) [reply lgtm.yml:43]: removed

**Comment by kcudnik** [comment]
not sure why latests builds dont pull lgtm.yml changes

**Review by nazariig — APPROVED** [review]

**Comment by nazariig** [comment]
@lguohan can you please help to merge?

**Comment by qiluo-msft** [comment]
Is it true that the failing LGTM log indicates it is not using 202012 branch? There is no `git clone -b 202012` as in your PR.

**Comment by kcudnik** [comment]
> Is it true that the failing LGTM log indicates it is not using 202012 branch? There is no `git clone -b 202012` as in your PR.

there is no specific clone for sonic-swss in lgtm.yml

**Review by prsunny — APPROVED** [review]

**Comment by kcudnik** [comment]
ping, c++ should start passing after it will this commit will be merged

**Pull request merged**